src/publish-build/index.ts

```diff
@@ -0,0 +1,97 @@
+import * as core from '@actions/core';
+import {
+  S3Client,
+  PutObjectCommand,
+  PutObjectCommandOutput,
+  PutObjectCommandInput,
+} from '@aws-sdk/client-s3';
+import { promises as fs, createReadStream } from 'fs';
+import mime from 'mime';
+import * as path from 'path';
+
+const REQUIRED_ENV_VARS = [
+  'AWS_BUCKET',
+  'PROJECT',
+  'SOURCE_DIR',
+  'DESTINATION_DIR',
+  'COMMIT_HASH',
+  'GITHUB_WORKSPACE',
+];
+
+interface envParams {
+  AWS_BUCKET: string;
+  PROJECT: string;
+  SOURCE_DIR: string;
+  DESTINATION_DIR: string;
+  COMMIT_HASH: string;
+  GITHUB_WORKSPACE: string;
+}
+
+function getEnvVars(): envParams {
+  const res: { [key: string]: string } = {};
```

Comment:
Could this type be `Partial<envParams>`?

Reply:
done!